utils/serviceClient.ts

```diff
@@ -0,0 +1,80 @@
+import * as te from "fp-ts/lib/TaskEither";
+
+import {
+  IResponseErrorInternal,
+  ResponseErrorInternal
+} from "@pagopa/ts-commons/lib/responses";
+
+import { getFetch } from "@pagopa/ts-commons/lib/agent";
+
+import { LimitedProfile } from "@pagopa/io-functions-commons/dist/generated/definitions/LimitedProfile";
+
+import { FiscalCode } from "@pagopa/ts-commons/lib/strings";
+
+import { toError } from "fp-ts/lib/Either";
+
+const fetch = getFetch(process.env);
+
+export interface IServiceClient {
+  readonly getLimitedProfileByPost: (
+    reqHeaders: NodeJS.Dict<string | ReadonlyArray<string>>,
+    fiscalCode: FiscalCode
+  ) => te.TaskEither<IResponseErrorInternal, LimitedProfile>;
+  readonly submitMessageForUser: (
+    reqHeaders: NodeJS.Dict<string | ReadonlyArray<string>>,
+    reqPayload: Response
+  ) => te.TaskEither<IResponseErrorInternal, Response>;
+}
+
+export const createClient = (
+  apiUrl: string,
+  apiKey: string
+): IServiceClient => ({
+  getLimitedProfileByPost: (
+    reqHeaders,
+    fiscalCode
+  ): ReturnType<IServiceClient["getLimitedProfileByPost"]> =>
+    te
+      .tryCatch(
+        () =>
+          fetch(`${apiUrl}/api/v1/profiles`, {
+            body: JSON.stringify({ fiscal_code: fiscalCode }),
+            headers: {
+              ...reqHeaders,
+              ["X-Functions-Key"]: apiKey
+            },
+            method: "POST"
+          }),
+        error => ResponseErrorInternal(String(error))
+      )
+      .chain(responseRaw =>
+        te.tryCatch(
+          () => responseRaw.json(),
+          error => ResponseErrorInternal(String(error))
+        )
+      )
+      .chain(response =>
```

Comment:
Add error handling in case `responseRaw.ok` is `false`
```suggestion
      .chain(responseRaw =>
        te.tryCatch(
          () => responseRaw.json(),
          error => ResponseErrorInternal(String(error))
        )
        .filterOrElseL(
            _ => responseRaw.ok,
            _ => ResponseErrorInternal(`Error calling client api: ${_}`)
          )
      )
      .chain(response =>
```

Reply:
I didn't really understand this comment. Isn't this scenario covered by the `tryCatch` around the fetch call?

Reply:
No, if the `responseRaw` return a not successfully status code, this is fine for him and it will be passed to the next chain. 
I tried it using `io-mock`

Reply:
For me to better understand: that's when the returned status code is different than 200? Isn't such scenario covered by the next payload decode?

Reply:
Testing it with mocke services, when the status code is different from 200, the next payload decode will manage it returning a validation error (wrapped inside a 500 error)

Reply:
ok